spirv_glsl.cpp

```diff
@@ -1041,14 +1047,32 @@ void CompilerGLSL::emit_header()
 		if (execution.flags.get(ExecutionModePostDepthCoverage))
 			inputs.push_back("post_depth_coverage");
 
+		if (execution.flags.get(ExecutionModePixelInterlockOrderedEXT) ||
+			execution.flags.get(ExecutionModePixelInterlockUnorderedEXT) ||
+			execution.flags.get(ExecutionModeSampleInterlockOrderedEXT) ||
+			execution.flags.get(ExecutionModeSampleInterlockUnorderedEXT))
+		{
+			statement("#if defined(GL_ARB_fragment_shader_interlock)");
+		}
+
 		if (execution.flags.get(ExecutionModePixelInterlockOrderedEXT))
-			inputs.push_back("pixel_interlock_ordered");
+			statement("layout(pixel_interlock_ordered) in;");
```

Comment:
I assume it's allowed to declare:
```
layout(pixel_interlock_ordered) in;
layout(early_fragment_tests) in;
```
where both execution modes are set?

Reply:
Yes, I've checked generated code, both modes are set in this case.